### PR TITLE
2239 text annotation alignment

### LINKED ISSUE #2239: Aligning Text Annotation with a Sequence flow with name, creates unhandled error in event listener
### Describe the Bug

An unhandled error in event listener happens when a Text Annotation is getting aligned with a Sequence flow that has a name
image

### Steps to Reproduce

1. Add a Sequence flow and add a name to it
2. Create an annotation for the flow
3. Now, try to align them in a vertical order (You need to release the mouse when the items are aligned)
image

4. The following error happens:
```
unhandled error in event listener TypeError: Failed to execute 'setTranslate' on 'SVGTransform': The provided float value is non-finite.
    at ro (SvgTransformUtil.js:35:13)
    at io.update (GraphicsFactory.js:277:5)
    at ChangeSupport.js:61:21
    at EventBus.js:614:13
    at Zr._invokeListener (EventBus.js:457:19)
    at Zr._invokeListeners (EventBus.js:431:24)
    at Zr.fire (EventBus.js:379:24)
    at ChangeSupport.js:46:14
    at EventBus.js:614:13
    at Zr._invokeListener (EventBus.js:457:19)
    at Zr._invokeListeners (EventBus.js:431:24)
    at Zr.fire (EventBus.js:379:24)
    at ChangeSupport.js:54:16
    at Array.forEach (<anonymous>)
    at ChangeSupport.js:53:14
    at EventBus.js:614:13
    at Zr._invokeListener (EventBus.js:457:19)
    at Zr._invokeListeners (EventBus.js:431:24)
    at Zr.fire (EventBus.js:379:24)
    at Kv._popAction (CommandStack.js:503:20)
    at Kv.execute (CommandStack.js:187:8)
    at Kg.moveElements (Modeling.js:243:22)
    at Move.js:177:14
    at EventBus.js:614:13
    at Zr._invokeListener (EventBus.js:457:19)
    at Zr._invokeListeners (EventBus.js:431:24)
    at Zr.fire (EventBus.js:379:24)
    at s (Dragging.js:187:21)
    at l (Dragging.js:283:21)
    at HTMLDocument.u (Dragging.js:332:5)
    at HTMLDocument.o (helpers.ts:98:17)
```

### Expected Behavior

No errors

### Environment

 - Browser: Chrome Version 128

### PR DESCRIPTION
**Pull request by abdul99ahad**
Closes #2239 

### Proposed Changes

Text annotation can now co exist with flow element label 

 

### Checklist

To ensure you provided everything we need to look at your PR:

* [x] **Brief textual description** of the changes present
* [ ] **Visual demo** attached
* [ ] **Steps to try out** present, i.e. [using the `@bpmn-io/sr` tool](https://github.com/bpmn-io/sr)
* [x] Related issue linked via `Closes {LINK_TO_ISSUE}` or `Related to {LINK_TO_ISSUE}`

Requires upstream: https://github.com/bpmn-io/diagram-js/pull/940

### CODE REVIEW

File: lib/features/modeling/behavior/AdaptiveLabelPositioningBehavior.js [hunk 1]
```diff
@@ -122,6 +122,10 @@ export default function AdaptiveLabelPositioningBehavior(eventBus, modeling) {
 
     var elementTrbl = asTRBL(element);
 
+    if (Object.values(elementTrbl).some(value => value == undefined || isNaN(value))) {
```
Comment (nikku) [inline lib/features/modeling/behavior/AdaptiveLabelPositioningBehavior.js:125]: Is this the root cause of the issue, or do we workaround a bug here?

Please add a test case to verify the behavior. I suggest you to do this as a start (to also guide your root causing efforts).

Comment (abdul99ahad) [reply lib/features/modeling/behavior/AdaptiveLabelPositioningBehavior.js:125]: The issue was that the flow element doesn't have top bottom coordinates which was returning NaN when updating the position of the element. I'll add test case for this.

**Comment by nikku** [comment]
Seems like tests fail on MacOS. This happens because elements have slightly different bounds across different OS; usually fixed by allocating additional space in the test case.

Can you have a look @abdul99ahad?

**Review by nikku — CHANGES_REQUESTED** [review]
Cf. previous comment. Test currently fails on MacOS.

**Comment by abdul99ahad** [comment]
> Seems like tests fail on MacOS. This happens because elements have slightly different bounds across different OS; usually fixed by allocating additional space in the test case.
> 
> Can you have a look @abdul99ahad?

Test is not failing on my macOS. @barmac can you confirm by running the test cases?

**Comment by abdul99ahad** [comment]
> Seems like tests fail on MacOS. This happens because elements have slightly different bounds across different OS; usually fixed by allocating additional space in the test case.
> 
> Can you have a look @abdul99ahad?

Updated. Test is now passing on all platforms. Ready for review.

File: test/spec/features/modeling/behavior/AdaptiveLabelPositioningBehaviorSpec.js [hunk 2]
```diff
@@ -224,6 +223,26 @@ describe('modeling/behavior - AdaptiveLabelPositioningBehavior', function() {
 
     });
 
+    describe('on annotation move / layout', function() {
```
Comment (nikku) [inline test/spec/features/modeling/behavior/AdaptiveLabelPositioningBehaviorSpec.js:226]:
```suggestion

    describe('on annotation move / layout', function() {
```

**Review by nikku — APPROVED** [review]
Good stuff, thanks for following up!

**Pull request merged**